packages/hub/services/scheduled-payments/executor.ts

```diff
@@ -51,6 +52,25 @@ export default class ScheduledPaymentsExecutorService {
     return gasPriceInToken(provider, gasTokenAddress);
   }
 
+  async executeScheduledPayments() {
+    // TODO: take a only a few, can be 10, to not overwhelm this method
+    // TODO: randomize the order
```

Comment:
It's unfortunate that Prisma API [doesn't support randomizing](https://github.com/prisma/prisma/discussions/5886#discussioncomment-2162265) the results. I suppose  we'll need to convert the implementation in ` fetchScheduledPayments` to raw sql to achieve this.

Reply:
I think we could order the scheduled payment by the last attempt.

Reply:
It looks like sorting by child records [is not supported](https://stackoverflow.com/questions/67930989/prisma-order-by-relation-has-only-count-property-can-not-order-by-relation-fie) by Prisma as well 🫠 I just tried adding `orderBy: { scheduledPaymentAttempts: { endedAt: 'asc' } },` and there's a type mismatch.

Reply:
Or it could be filtered after fetching to the database.

Reply:
You mean by loading all the scheduled payments that are due and then filtering/ordering? That could work while we have a low number of scheduled payments